### PR TITLE
Profiles: improve registrations fetch

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes RNBW-3411

Fixes RNBW-3412

## What changed (plus any additional context for devs)

Improving registration fetch

- images are now cached, before we were querying for avatar and cover images all the time

before: fetchImages was being called every time `fetchAccountENSDomains` is called, even tho that fetch was being cached 
https://recordit.co/L8os83EF1R

now we're saving those values to avoid unnecessary requests
https://recordit.co/F4ZwZ8idxC

- saving registration to local storage so we can use it as default while getting all the ens names of a wallet in the intro screen

in the video you can see at the beginning the data used to load the screen is the one saved in local storage, then data comes from subgraph and we use it instead
https://recordit.co/f8Pk5AWS7U

## Final checklist

- [ ] Assigned individual reviewers?
- [ ] Added labels?
- [ ] Added e2e tests? if not please specify why
- [ ] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

File: src/hooks/useAccountENSDomains.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
```
Comment (jxom) [inline src/hooks/useAccountENSDomains.tsx:1]: Is this file meant to be `.tsx`?

File: src/hooks/useAccountENSDomains.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    async () => await fetchImages(name),
+    {
+      staleTime: 120000,
+    }
+  );
+}
+
+export default function useAccountENSDomains() {
+  const { accountAddress } = useAccountSettings();
+  const [initialENSDomains, setIinitialENSDomains] = useState<
```
Comment (jxom) [inline src/hooks/useAccountENSDomains.tsx:67]:
```suggestion
  const [initialENSDomains, setInitialENSDomains] = useState<
```

**Comment by estebanmino** [comment]
https://user-images.githubusercontent.com/12115171/165215403-e97f42e1-0472-459c-8de4-1504881d8387.mp4

how fast

**Review by jxom — APPROVED** [review]
dope

**Pull request merged**